From: Richard Goedeken <Richard@fascinationsoftware.com>
Date: Wed, 12 Aug 2015 14:26:57 -0700
Subject: add OUYA wireless controller auto-config from github ticket #600

Origin: upstream, https://github.com/mupen64plus/mupen64plus-input-sdl/commit/67b4aa7a33421c3bb6645c5e3e221efc1d392e00
---
 data/InputAutoCfg.ini | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/data/InputAutoCfg.ini b/data/InputAutoCfg.ini
index 99a913e..9b410b4 100644
--- a/data/InputAutoCfg.ini
+++ b/data/InputAutoCfg.ini
@@ -922,6 +922,29 @@ Rumblepak switch =
 X Axis = axis(0-,0+)
 Y Axis = axis(1-,1+)
 
+[OUYA Game Controller]
+plugged = True
+plugin = 2
+mouse = False
+Mempak switch =
+Rumblepak switch =
+DPad U = button(8)
+DPad D = button(9)
+DPad L = button(10)
+DPad R = button(11)
+Start = button(7)
+Z Trig = button(12)
+A Button = button(0)
+B Button = button(1)
+C Button U = axis(4-)
+C Button D = axis(4+)
+C Button L = axis(3-)
+C Button R = axis(3+)
+L Trig = button(4)
+R Trig = button(5)
+X Axis = axis(0-,0+)
+Y Axis = axis(1-,1+)
+
 [PC Game Controller]
 plugged = True
 plugin = 2
