From: Richard Goedeken <Richard@fascinationsoftware.com>
Date: Fri, 12 Jun 2015 21:30:05 -0700
Subject: add MayFlash PC048 adapter auto-config from Brandon Bundy via the
 google group

Origin: upstream, https://github.com/mupen64plus/mupen64plus-input-sdl/commit/198b8a7f03d4ecc3fd941ce2cf08066324771fab
---
 data/InputAutoCfg.ini | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/data/InputAutoCfg.ini b/data/InputAutoCfg.ini
index f128e3b..2569f89 100644
--- a/data/InputAutoCfg.ini
+++ b/data/InputAutoCfg.ini
@@ -1483,6 +1483,32 @@ Rumblepak switch = key(114)
 Y Axis = axis(1-,1+)
 X Axis = axis(0-,0+)
 
+; MayFlash PC048 adapter
+[PS/SS/N64 Joypad to USB BOX]
+plugged = True
+plugin = 2
+mouse = False
+AnalogDeadzone = 1024,1024
+AnalogPeak = 32768,32768
+DPad R = hat(0 Right)
+DPad L = hat(0 Left)
+DPad D = hat(0 Down)
+DPad U = hat(0 Up)
+Start = button(8)
+Z Trig = button(9)
+B Button = button(2)
+A Button = button(1)
+C Button R = axis(2-)
+C Button L = axis(2+)
+C Button D = axis(3+)
+C Button U = axis(3-)
+R Trig = button(7)
+L Trig = button(5)
+Mempak switch =
+Rumblepak switch =
+Y Axis = axis(1-,1+)
+X Axis = axis(0-,0+)
+
 ; Asikgame - Super Dual Box (with two PS2 gamepads connected)
 ; this is a 2-port USB device. The Right port is controller #1, the Left port is #2
 [USB Human(2p) Interface Device]
